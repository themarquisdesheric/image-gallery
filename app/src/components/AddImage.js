import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MdAddCircleOutline from 'react-icons/lib/md/add-circle-outline';

class AddImage extends Component {
  constructor(props) {
    super(props);

    this.state = this.getInitialState();

    this.handleChange = this.handleChange.bind(this);
  }

  static propTypes = { onAdd: PropTypes.func.isRequired }

  getInitialState() {
    return {
      title: '',
      description: '',
      url: ''
    };
  }

  reset() {
    this.setState(this.getInitialState());
  }

  handleChange(event) {
    const { name, value } = event.target;

    this.setState({
      [name]: value
    });
  }

  render() {
    const { title, description, url } = this.state;

    return (
      <div className="Add">
        <form onSubmit={e => {
          e.preventDefault();
          this.props.onAdd(this.state);
          this.reset();
        }}>
          <fieldset>
            <legend>
              <h2 className="Add-header">
                <MdAddCircleOutline className="Add-icon" />
                Add an Image
              </h2>
            </legend>
            <label>
              Title:
                <input 
                  name="title" 
                  value={title}
                  onChange={this.handleChange}
                />
            </label>
            <label>
              Description:
                <input
                  name="description"
                  value={description}
                  onChange={this.handleChange}
                />
            </label>
            <label>
              URL:
                <input
                  name="url"
                  value={url}
                  onChange={this.handleChange}
                />
            </label>
            <button type="submit">Add</button>
          </fieldset>
        </form>
      </div>
    );
  }
}

export default AddImage;