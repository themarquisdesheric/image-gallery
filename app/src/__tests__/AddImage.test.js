import React from 'react';
import renderer from 'react-test-renderer';
import AddImage from '../components/AddImage';

function handleAddImage() {}

describe('AddImage', () => {
  it('renders a form to add images', () => {
    const tree = renderer.create(<AddImage onAdd={handleAddImage} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});