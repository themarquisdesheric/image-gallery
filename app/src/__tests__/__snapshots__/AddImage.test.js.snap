// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddImage renders a form to add images 1`] = `
<div
  className="Add"
>
  <form
    onSubmit={[Function]}
  >
    <fieldset>
      <legend>
        <h2
          className="Add-header"
        >
          <svg
            className="Add-icon"
            fill="currentColor"
            height="1em"
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "color": undefined,
                "verticalAlign": "middle",
              }
            }
            viewBox="0 0 40 40"
            width="1em"
          >
            <g>
              <path
                d="m20 33.4q5.5 0 9.4-4t4-9.4-4-9.4-9.4-4-9.4 4-4 9.4 4 9.4 9.4 4z m0-30q6.9 0 11.8 4.8t4.8 11.8-4.8 11.8-11.8 4.8-11.8-4.8-4.8-11.8 4.8-11.8 11.8-4.8z m1.6 8.2v6.8h6.8v3.2h-6.8v6.8h-3.2v-6.8h-6.8v-3.2h6.8v-6.8h3.2z"
              />
            </g>
          </svg>
          Add an Image
        </h2>
      </legend>
      <label>
        Title:
        <input
          name="title"
          onChange={[Function]}
          value=""
        />
      </label>
      <label>
        Description:
        <input
          name="description"
          onChange={[Function]}
          value=""
        />
      </label>
      <label>
        URL:
        <input
          name="url"
          onChange={[Function]}
          value=""
        />
      </label>
      <button
        type="submit"
      >
        Add
      </button>
    </fieldset>
  </form>
</div>
`;
